File: 1014-k-closest-points-to-origin/1014-k-closest-points-to-origin.py
class Solution:
    def distanceFromOrigin(self, a) -> int:
        return (a[0] * a[0]) + (a[1] * a[1])

    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        return sorted(points, key= lambda x : self.distanceFromOrigin(x))[:k]
        

File: 1014-k-closest-points-to-origin/1014-k-closest-points-to-origin.cpp
class Solution {
public:
    static int distance(const std::vector<int>& a) {
        return a[0] * a[0] + a[1] * a[1];
    }

    static bool compare(vector<int> a, vector<int> b) {
        return distance(a) < distance(b);
    }

    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        sort(points.begin(), points.end(), compare);

        vector<vector<int>> nearestPoints;
        
        for (int i = 0; i < k; i++) {
            nearestPoints.push_back(points[i]);
        }

        return nearestPoints;
    }
};